src/Core/BackgroundSchedulePool.cpp

```diff
@@ -305,7 +305,10 @@ void BackgroundSchedulePool::threadFunction()
         }
 
         if (task)
+        {
             task->execute();
+            DB::QuillLogger::reset_thread_context();
```

Comment:
Same question about `reset_thread_context` here. Please leave a comment explaining what it does and why it's necessary.